conftest.py

```diff
@@ -111,156 +261,72 @@ def temp_dir():
 
 
 @pytest.fixture(scope="class")
-def answer_file(request):
-    r"""
-    Assigns the name of the appropriate answer file as an attribute of
-    the calling answer test class.
+def ds(request):
+    # data_dir_load can take the cls, args, and kwargs. These optional
+    # arguments, if present,  are given in a dictionary as the second
+    # element of the list
+    if isinstance(request.param, typing.Sequence):
+        ds_fn, opts = request.param
+    else:
+        ds_fn = request.param
+        opts = {}
+    try:
+        return data_dir_load(
+            ds_fn, cls=opts.get("cls"), args=opts.get("args"), kwargs=opts.get("kwargs")
+        )
+    except FileNotFoundError:
+        return pytest.skip(f"Data file: `{request.param}` not found.")
 
-    The answer file is the file that either already contains previously
-    generated answers or else is going to contain newly generated
-    answers. If an answer file already exists, then it cannot be used
-    to hold newly generated answers; a fresh file must be used instead.
 
-    Each class that performs answer tests (e.g., TestEnzo) has a
-    corresponding answer file. These answer files are stored in
-    test_data_dir/answers. The names of the files and their
-    corresponding answer class are in ./tests/tests.yaml.
+@pytest.fixture(scope="class")
+def field(request):
+    """
+    Fixture for returning the field. Needed because indirect=True is
+    used for loading the datasets.
```

Comment:
Why would indirect=True make it so fields aren't accessible?

Reply:
I'm not sure, to be honest. All I know is that things did not work correctly unless the dataset parameterizations used `indirect=True` and as soon as I did that, the other fixtures stopped working until I made them use `indirect=True`, as well.